api/model/src/main/java/org/projectnessie/model/CommitResponse.java

```diff
@@ -61,6 +61,24 @@ default Map<ContentKey, String> toAddedContentsMap() {
     return added.stream().collect(Collectors.toMap(AddedContent::getKey, AddedContent::contentId));
   }
 
+  /**
+   * If new content has been added to Nessie, updates the {@link Content#getId() content ID} of the
+   * given content with the generated content ID.
+   */
+  @SuppressWarnings("unchecked")
+  default <T extends Content> T contentWithId(ContentKey key, T content) {
+    List<AddedContent> addedContents = getAddedContents();
+    if (addedContents == null) {
+      return content;
+    }
+    return (T)
+        getAddedContents().stream()
+            .filter(added -> added.getKey().equals(key))
+            .findFirst()
+            .map(added -> content.withId(added.contentId()))
```

Comment:
Not sure I understand the purpose of this method :thinking: why not simply return `added` to the caller?

Reply:
right :facepalm: